ui: statements tracing - Redux API

koorosh opened the pull request:
Depends on: #45274 

- [ ] Following commit has to be removed before merge: https://github.com/cockroachdb/cockroach/pull/45494/commits/14fc8e79e1b57e07e8c8227cc0e380afe45cc28a

- [ ] Has to be merged after backend work is done for Statement tracing API (PR: #45533)

Statements sagas are implemented to be able request
statement diagnostics and retrieve when it's ready.

This commit defines Actions for requesting diagnostics
and sagas which listens for those actions.

Also selectors are added to find statement by
statement id (query) which contains diagnostic data.

koorosh committed:
ui: (fix) Apply styles for navigation panes

Before, Text component styles were adjusted to style
links on side navigation panel. It was wrong because
Text component should not care about particular styling
of navigation pane.
Now, particular styles which relate to navigation panel
only is moved to SideNavigation component.
It is changed now because current implementation breaks
styles for Diagnostics panel.

Release note: None

koorosh committed:
ui: Link component

Link component is a wrapper on top of <a> element
is created to maintain consistent styles for links.

Release note: None

koorosh committed:
ui: Add Diagnostics view for Statements

- Adjust SummaryCard to provide external styles
- Add DiagnosticsView component which renders
view when no diagnostics available.
- Current implementation includes only view without
any logic connected to redux store.

Release note: None

koorosh committed:
ui: Add sagas to handle statement tracing actions

Statements sagas are implemented to be able request
statement diagnostics and retrieve when its ready.

This commit defines Actions for requesting diagnostics
and sagas which listens for those actions.

Also selectors are added to find statement by
statement id (query) which contains diagnostic data.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45494)
<!-- Reviewable:end -->

dhartunian committed:
DELETE!!! add API for accessing statement diagnostics

Added 3 new RPC endpoints under `_status` for manipulating Statement Diagnostics
Requests:

1. `CreateStatementDiagnosticsRequest`
Creates a new request for a given statement fingerprint

2. `StatementDiagnosticsRequests`
Fetches all statement diagnostics requests

3. `StatementDiagnostics`
Fetches a specific statement diagnostics request that has been completed and
includes the JSON payload of the trace

Basic implementations for these endpoints have been added to the
`apiReducers.tsx` and `api.ts` files for use by the Admin UI.

A minor bug where completed requests did not have `completedAt` timestamp set
was fixed.

Release note (<category, see below>): <what> <show> <why>

koorosh committed:
ui: Rename Link component to Anchor

Release note: None

koorosh committed:
ui: Integrate requestStatementDiagnostics API

Replace faked API with real implementation to
request statement diagnostics.

Release note: None

koorosh committed:
ui: Add selectDiagnosticRequestsByStatementFingerprint selector

Release note: None

koorosh committed:
ui: Connect Statement Diagnostics API

Before, Statements diagnostics view used faked API
to simulate responses from server.
Now redux API has changed to comply with actual
implementation of API.

Release note: None

koorosh commented:
Outdated
